app/views/users/edit.html.erb

```diff
@@ -0,0 +1,29 @@
+<% provide(:title, "Edit user") %>
+<h1>Update your profile</h1>
+
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_with(model: @user) do |f| %>
```

Comment:
form_with, form_for, form_tag. What's the difference between them?

Reply:
**form_with** vs **form_for** vs **form_tag**

1. **form_with:**

- luôn sử dụng form builder
- must specify all your ids and classes manually
- form_with submits are remote by default (data-remote='true') --> can be changed by using local: true
- when passed a model --> form_with works like form_for
- when not passed a model → form_with works like form_tag
- is an update from form_for & form_tag → it is suggested to use this

2. **form_for:**

- tạo form khi có một model cụ thể --> model’s attributes to build create or update form
- generates automatic ids for the input fields
- form_for @user == form_with model: @user

3. **form_tag:**

- tạo form khi k có model cơ bản --> creates an HTML form tag with the passed URL as action
- generates automatic ids for the input fields